src/wmtk/operations/tet_mesh/TetSplit.cpp

```diff
@@ -0,0 +1,91 @@
+#include "TetSplit.hpp"
+#include <spdlog/spdlog.h>
+#include <wmtk/SimplicialComplex.hpp>
+#include <wmtk/invariants/ValidTupleInvariant.hpp>
+#include <wmtk/invariants/find_invariant_in_collection_by_type.hpp>
+
+namespace wmtk::operations {
+
+void OperationSettings<tet_mesh::TetSplit>::initialize_invariants(const TetMesh& m)
+{
+    // outdated + is valid tuple
+    invariants = basic_invariant_collection(m);
+}
+
+bool OperationSettings<tet_mesh::TetSplit>::are_invariants_initialized() const
+{
+    return find_invariants_in_collection_by_type<ValidTupleInvariant>(invariants);
+}
+
+namespace tet_mesh {
+
+// TetSplit::TetSplit(Mesh& m, const Tuple& t, const OperationSettings<TetSplit>& settings)
+//     : TetMeshOperation(m)
+//     , TupleOperation(settings.invariants, t)
+//     , m_settings{settings}
+// {
+//     assert(m_settings.are_invariants_initialized());
+// }
+
+TetSplit::TetSplit(TetMesh& m, const Tuple& t, const OperationSettings<TetSplit>& settings)
+    : TetMeshOperation(m)
+    , TupleOperation(settings.invariants, t)
+    , m_settings{settings}
+{
+    assert(m_settings.are_invariants_initialized());
+}
+
+bool TetSplit::execute()
+{
+    Tuple first_split_ret = mesh().split_edge(input_tuple(), hash_accessor()).m_output_tuple;
```

Comment:
You should not use atomic operations in here but `TetEdgeSplit` and `TetEdgeCollapse`.

Also, check the naming convention from the tri_mesh folder. `TetEdgeSplit` should only be called `EdgeSplit` and `TetEdgeCollapse` should be `EdgeCollapse`.